src/main/features/core/windowTitleInfo.js

```diff
@@ -0,0 +1,16 @@
+const windowTitle = WindowManager.get(global.mainWindowID).getTitle();
+let songInfo = {};
+
+if (Settings.get('enableWindowTitleInfo')) {
+  PlaybackAPI.on('change:song', (_songInfo) => {
+    songInfo = _songInfo;
+    WindowManager.get(global.mainWindowID).setTitle(
+      (_songInfo.title && _songInfo.artist) ? `${songInfo.title} - ${songInfo.artist}` : windowTitle
```

Comment:
They won't be null, but they can be empty. Just figured that one out. I have a few tracks without artist names, but if its that far out as a use case I'll just remove the check.